### PR TITLE
Added Fractional Knapsack

### LINKED ISSUE #170: [Feature Request] Addition of Fractional Knapsack under greedy algorithms
### Describe the solution you'd like

Adding a file in the Greedy Algorithms detailing the Fractional Knapsack Problem, including the problem statement, the greedy approach to solve it (with value-to-weight ratio as the heuristic), and an example to show how the algorithm works step-by-step. This would improve the completeness of the greedy algorithm coverage.

### Additional context
Including the problem with a simple code snippet (preferably in Python or any supported language in the repo) would enhance understanding.
---

**Would you like to work on this feature?**  
- [x] Yes  
- [ ] No

### PR DESCRIPTION
**Pull request by Himanshi-m**
### Description
Added a file in the Greedy Algorithms section covering the Fractional Knapsack Problem. The file includes the problem statement, explains the greedy approach using the value-to-weight ratio as the heuristic, and provides a step-by-step example of how the algorithm works. This improves the overall coverage of greedy algorithms by offering a detailed explanation and practical illustration of the fractional knapsack method.

Fixes #170 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/greedy-algorithms/fractional-knapsack.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+### Problem Statement:
+
+You are given `N` items, each with a given weight `w_i` and value `v_i`, and a knapsack with a maximum capacity `W`. The objective is to maximize the total value of the items placed in the knapsack. However, in the Fractional Knapsack Problem, you are allowed to take fractions of an item. Your goal is to find the optimal way to fill the knapsack to maximize the total value without exceeding the capacity `W`.
+
+### Time Complexity:
+
+- **Best, Average, and Worst Case: O(N log N)**  
+The dominant operation is sorting the items based on their value-to-weight ratio, which takes O(N log N), where N is the number of items.
+
```
Comment (ajay-dhangar) [inline docs/greedy-algorithms/fractional-knapsack.md]: replace `O(N log N)` with `$O(N log N)$`

File: docs/greedy-algorithms/fractional-knapsack.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+### Space Complexity:
+
+- **Space Complexity: O(N)**  
+Space is required for the list of items and auxiliary data structures for sorting, resulting in O(N) space complexity.
+
```
Comment (ajay-dhangar) [inline docs/greedy-algorithms/fractional-knapsack.md]: replace `O(N)` with `$O(N)$`

File: docs/greedy-algorithms/fractional-knapsack.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    double maxValue = fractionalKnapsack(W, items);
+    cout << "Maximum value we can obtain: " << maxValue << endl;
+
+    return 0;
+}
+```
+
+### Summary:
+
+The Fractional Knapsack Problem is a classic example of a greedy algorithm. It efficiently maximizes the value of the items that can be carried in the knapsack by selecting items based on their value-to-weight ratio. This algorithm is especially useful when fractional items can be taken and provides the optimal solution with a time complexity of O(N log N). The fractional knapsack finds applications in resource allocation, budget management, and similar optimization problems.
```
Comment (ajay-dhangar) [inline docs/greedy-algorithms/fractional-knapsack.md]: replace `O(N log N)` with `$O(N log N)$`

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions for best practices

**Comment by Himanshi-m** [comment]
Updated. @ajay-dhangar please review.

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**